python/lsst/ts/standardscripts/maintel/closed_loop.py

```diff
@@ -0,0 +1,519 @@
+from lsst.ts.observatory.control.maintel.mtcs import MTCS, MTCSUsages
+from lsst.ts.observatory.control.remote_group import Usages
+
+STD_TIMEOUT = 10
+
+
+# TODO - Subsitute by class in idl.enum when available
+class ClosedLoopMode(enum.IntEnum):
+    CWFS = 0
+    FAM = 1
+
+
+class DOFComponent(enum.IntEnum):
+    M2_dz = 0
+    M2_dx = 1
+    M2_dy = 2
+    M2_rx = 3
+    M2_ry = 4
+    Cam_dz = 5
+    Cam_dx = 6
+    Cam_dy = 7
+    Cam_rx = 8
+    Cam_ry = 9
+    M1M3_B1 = 10
+    M1M3_B2 = 11
+    M1M3_B3 = 12
+    M1M3_B4 = 13
+    M1M3_B5 = 14
+    M1M3_B6 = 15
+    M1M3_B7 = 16
+    M1M3_B8 = 17
+    M1M3_B9 = 18
+    M1M3_B10 = 19
+    M1M3_B11 = 20
+    M1M3_B12 = 21
+    M1M3_B13 = 22
+    M1M3_B14 = 23
+    M1M3_B15 = 24
+    M1M3_B16 = 25
+    M1M3_B17 = 26
+    M1M3_B18 = 27
+    M1M3_B19 = 28
+    M1M3_B20 = 29
+    M2_B1 = 30
+    M2_B2 = 31
+    M2_B3 = 32
+    M2_B4 = 33
+    M2_B5 = 34
+    M2_B6 = 35
+    M2_B7 = 36
+    M2_B8 = 37
+    M2_B9 = 38
+    M2_B10 = 39
+    M2_B11 = 40
+    M2_B12 = 41
+    M2_B13 = 42
+    M2_B14 = 43
+    M2_B15 = 44
+    M2_B16 = 45
+    M2_B17 = 46
+    M2_B18 = 47
+    M2_B19 = 48
+    M2_B20 = 49
+
+
+class ClosedLoop(salobj.BaseScript, metaclass=abc.ABCMeta):
+    """Closed loop script. This script is used to perform measurements of
+    the wavefront error, then propose dof offsets based on ts_ofc.
+
+    Parameters
+    ----------
+    index : `int`, optional
+        Index of Script SAL component (default=1).
+    remotes : `bool`, optional
+        Should the remotes be created (default=True)? For unit testing this
+        can be set to False, which allows one to mock the remotes behaviour.
+    descr : `str`, optional
+        Short description of the script.
+
+    Notes
+    -----
+    **Checkpoints**
+
+    - "Taking image...": If taking in-focus detection image.
+    - "[N/MAX_ITER]: CWFS loop starting...": Before each cwfs iteration, where
+        "N"  is the iteration number and "MAX_ITER" is the maximum number of
+        iterations.
+    - "[N/MAX_ITER]: CWFS converged.": Once CWFS reaches convergence.
+    - "[N/MAX_ITER]: CWFS applying correction.": Just before
+        corrections are applied.
+
+    **Details**
+
+    This script is used to perform measurements of the wavefront error, then
+    propose dof offsets based on ts_ofc. The offsets are not applied
+    automatically and must be turned on by the user through
+    apply_corrections attribute.
+    """
+
+    def __init__(self, index=1, add_remotes=True, descr=""):
+        super().__init__(
+            index=index,
+            descr=descr,
+        )
+
+        self.mtaos = RemoteGroup(
+            domain=self.domain,
+            components=["MTAOS"],
+            intended_usage=None if add_remotes else Usages.DryTest,
+            log=self.log,
+        )
+
+        mtcs_usage = None if add_remotes else MTCSUsages.DryTest
+        self.mtcs = MTCS(self.domain, intended_usage=mtcs_usage, log=self.log)
```

Comment:
I think the more recent approach to this is setting values for these to `None` here and postponing the creating to the configuration stage. 

See how we did this in the  [move_p2p](https://github.com/lsst-ts/ts_standardscripts/blob/develop/python/lsst/ts/standardscripts/maintel/move_p2p.py) script.